File: src/market-tracker-webapi/Application/Domain/Models/Market/Retail/Shop/Store.cs
namespace market_tracker_webapi.Application.Domain.Models.Market.Retail.Shop;

public record Store(StoreId Id, string Name, string Address, City? City, Company Company, Guid OperatorId)
{
    public Store(
        int Id,
        string Name,
        string Address,
        City? City,
        Company Company,
        Guid OperatorId
    ) : this(
        new StoreId(Id),
        Name,
        Address,
        City,
        Company,
        OperatorId
    )
    {
    }


    public bool IsOnline => City is null;
}

public record StoreItem(int Id, string Name, string Address, int? CityId, int CompanyId, Guid OperatorId);

public record StoreId(
    int Value
);

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/domain/model/market/Store.kt
package pt.isel.markettracker.domain.model.market

data class Store(
    val id: Int,
    val name: String,
    val address: String,
    val city: City?,
    val company: Company,
    val isOnline: Boolean
) {
    fun toStoreInfo() = StoreInfo(id, name, address, city, isOnline)
}

data class StoreInfo(
    val id: Int,
    val name: String,
    val address: String,
    val city: City?,
    val isOnline: Boolean
)